client/test_data/up_or_down_turn5.json

```diff
@@ -103,7 +103,9 @@
         "turnDrawn": 0,
         "turnDiscarded": -1,
         "turnPlayed": -1,
-        "isMisplayed": false
+        "isMisplayed": false,
+        "matchingCards": [],
```

Comment:
well, this invalidates our current tests :/ Is it possible to update the script under test_data to generate valid tests that match your rewrite?

Reply:
Well I looked at that script. if I understood correctly, it just copies the data from the old logic. Since the new logic has more data (especially the separation between matchingCards and unseenCards), I don't think it's possible (without copy the whole algorithm).
Of course, I can change it so it will copy the data from the new logic, but then the tests won't verify that the behavior is the same as in the old code...
I think that observed behavior (pipStates, location, etc) should be compared to the old logic, but internal variables (unseenCard, matchingCards, matchingCardsArray) shouldn't. Hence, I suggest that I will change the script to match the new logic, and will verify that the only changes (in the current tests) are in the new variables.

Reply:
I implemented those changes.
I suggest again, that maybe we do want the state to have something similar to pipStates. Right now we recalculate the pip states of every card when a card is drawn (except of cards in the same hand). (And this is a ui calculation, so we can't put it in a reducer)